### PR TITLE
Remove extraneous string escape from convert to template string refactor

### LINKED ISSUE #37330: Convert to template string incorrectly adds escaped backslashes
*TS Template added by @mjbvz*

**TypeScript Version**: 3.8.3

**Search Terms**
- refactor
- refactoring

---

- VSCode Version: 1.43.0
- OS Version: Windows 7 Pro

Steps to Reproduce:

1.  Add the following text to a typescript file (have not tried javascript):
```
return "\\[[" + text + "](" + link + ")\\]";
```
2. Choose refactor -> Convert to template string
3. Observe the following output:
```
return `\\\\[[${text}](${link})\\\\]`;
```

This contains too many backslashes - there should only be one escaped slash in each location, i.e.

```
return `\\[[${text}](${link})\\]`;
```

Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes #37330

String literal `text` should be transferred to template head/tail `text` without modification—it gets escaped in the printer.

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -3465,8 +3465,9 @@ namespace ts {
     }
 
     function getReplacement(c: string, offset: number, input: string) {
-        if (c.charCodeAt(0) === CharacterCodes.nullCharacter) {
-            const lookAhead = input.charCodeAt(offset + c.length);
+        const charCode = c.charCodeAt(0);
```
Comment (sandersn) [inline src/compiler/utilities.ts:3468]: this extract variable doesn't seem necessary to me

Comment (andrewbranch) [reply src/compiler/utilities.ts:3468]: You’re right, I had initially made other changes in this function.

File: src/compiler/utilities.ts [hunk 2]
```diff
@@ -3465,8 +3465,9 @@ namespace ts {
     }
 
     function getReplacement(c: string, offset: number, input: string) {
-        if (c.charCodeAt(0) === CharacterCodes.nullCharacter) {
-            const lookAhead = input.charCodeAt(offset + c.length);
+        const charCode = c.charCodeAt(0);
+        if (charCode === CharacterCodes.nullCharacter) {
+            const lookAhead = input.charCodeAt(offset + 1);
```
Comment (andrewbranch) [inline src/compiler/utilities.ts:3470]:
```suggestion
        if (c.charCodeAt(0) === CharacterCodes.nullCharacter) {
            const lookAhead = input.charCodeAt(offset + c.length);
```

**Comment by andrewbranch** [comment]
@orta, playwright failed to install all of a sudden on node 8 both in Actions and in Pipelines. Have you seen this before? 🤔

**Comment by orta** [comment]
No, this sounds like the first time it's failed full stop

**Comment by orta** [comment]
Because the dep isn't locked, it looks like it needs bumping to node 10

**Comment by thw0rted** [comment]
I can't tell if this is supposed to have landed in production yet.  I'm running VSCode 1.49.0, which I think is TS 4.0 under the hood.  Should I expect the behavior I'm seeing to include this PR?

I came here because I just converted `"\b" + query  + "\b"` to a template and got "missing character" unicode boxes, i.e. `` `${query}` ``, as the result.

The missing character I pasted in there looks like an empty square in the GH Markdown editor but a question-mark inside a black diamond in the preview.  I copied the character, then pasted into the Chrome dev tools console as `"{paste}".charCodeAt(0)`, which returns `8` -- so it's converting the nice readable backspace escape sequence into a literal backspace character.

Obviously, in my case I actually meant to write `"\\b" + query + "\\b"` -- it was an oversight in the original code.  But the TS convert-to-template refactoring feature is changing the escape sequence to a literal.  I checked and this happens for all escape sequences: newline, form-feed, etc.  I'm pretty sure it shouldn't, and I'm trying to figure out if it was fixed by this PR, or if the PR introduced the bug.

I guess what I'm asking is, should I open a new issue?

**Comment by andrewbranch** [comment]
@thw0rted yes, if you could open a new issue, that would be great. Thank you!

**Pull request merged**